src/server/system_services/bucket_server.js

```diff
@@ -589,7 +589,7 @@ function get_bucket_changes_versioning(req, bucket, update_request, single_bucke
     if (update_request.versioning === 'DISABLED') {
         throw new RpcError('BAD_REQUEST', 'Cannot set versioning to DISABLED');
     }
-    if (bucket.namespace) {
+    if (bucket.namespace && !bucket.namespace.write_resource.resource.nsfs_config.fs_root_path) {
         throw new RpcError('BAD_REQUEST', 'Cannot set versioning on namespace buckets');
```

Comment:
looks like you need to update the error, we will be able to set versioning on nsfs

Reply:
updated